### PR TITLE
add zocl dma operation programming guide doc

### PR DESCRIPTION
**Pull request by xdavidz**
Please take a look at the changed doc and let me know if we should combine zocl_ioctl and zocl other design docs. Thank you so much for your time!
file:///proj/xsjhdstaff2/davidzha/XRT_zocl/build/Release/runtime_src/doc/html/zocl_design.html

This was the clone of original PR, due to merge conflicts, I re-generated the doc.
https://github.com/Xilinx/XRT/pull/1981

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_dma.h [hunk 1]
```diff
@@ -36,7 +59,29 @@ typedef struct zocl_dma_handle {
...
+ * zocl_dma_memcpy_pre() - DMA memory copy preparation.
+ *
+ * @dma_handle: zocl dma handler
+ * @dst_paddr: destination dma address
+ * @src_paddr: source dma address
+ * @size: number of bytes to read from source
+ *
+ * Caller get dma_handle from sched_cmd, then get dma_chan from
+ * zocl_dma_channel_instance; caller can optionally set dma_func and dma_arg for
+ * private usage.
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_dma.h]: It seems caller does not have option to set dma_func and dma_arg, right? We have to modify the function itself.

Comment (xdavidz) [reply src/runtime_src/core/edge/drm/zocl/zocl_dma.h]: What about caller can optionally make its own dma_func and dma_arg for private usage. Set implies there is existing dma_func, but actually caller has to make the dma_func and know what is inside the dma_arg.

File: src/runtime_src/core/edge/drm/zocl/zocl_dma.h [hunk 2]
```diff
@@ -45,9 +45,10 @@ typedef void (*zocl_dma_complete_cb)(void *arg, int ret);
  * @dma_func: call back function when dma complete
  * @dma_arg: private data for dma_func call back function
  *
- * Caller can set dma_func and dma_arg for handling subsequent asynchronous
- * operations, when DMA is completed dma_func will be called with dma_arg. DMA
- * errors is being returned via zocl_dma_complete_cb argument ret.
+ * Prior to pass this handler to ZOCL DMA Engine, user can set dma_func and
+ * dma_arg for handling subsequent asynchronous operations, when DMA is
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_dma.h]: ...operations. When DMA is ...

**Comment by xdavidz** [comment]
Removed rst files because the zocl dma engine doc is for programmer only. We should add doc to indicate that copyBo utilizes dma for better performance.

**Comment by david2011qq** [comment]
close it and merge with other PR

**Pull request closed without merging**